win32ss/user/ntuser/window.c

```diff
@@ -2190,16 +2190,6 @@ co_UserCreateWindowEx(CREATESTRUCTW* Cs,
 
    Window->rcClient = Window->rcWindow;
 
-   /* Link the window */
-   if (NULL != ParentWindow)
-   {
-      /* Link the window into the siblings list */
-      if ((Cs->style & (WS_CHILD|WS_MAXIMIZE)) == WS_CHILD)
-          IntLinkHwnd(Window, HWND_BOTTOM);
-      else
-          IntLinkHwnd(Window, hwndInsertAfter);
-   }
-
```

Comment:
Note: Below, in (new) lines 2203 and 2206 (just below the `IntIsTopLevelWindow` check), it is suspicious to use `Window->spwndParent` instead of the original `ParentWindow`, because the window hook made above (lines around 2131) might have changed it by calling `SetParent` too early (either because the user-mode application is buggy or because it wants to trigger an error in win32k).
Also we see that code just above (e.g. lines 2180 and down) still uses the original known `ParentWindow` even after said hook.

Reply:
It therefore may be worth somewhere to add a check to see whether `Window` has been linked too early and if so, at least now for diagnosis purposes, have some sort of DPRINT 🤔

Reply:
Excuse me, but how can I check the linking?

Reply:
With some code similar to this:
```c
    if (Window->spwndNext || Window->spwndPrev)
    {
        ERR("Window 0x%p has been linked too early!\n", Window);
    }
```

Reply:
Added the check.